### PR TITLE
create custom input and change all inputs to custom

### PR DESCRIPTION
**Pull request by juthikahoque**
## Notion ticket link

[Create Custom Input Component](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)

## Implementation description
* created a custom input component under common components
* added forwardRef so that we can pass in the ref prop needed for attaching a new file in CourtDetailsForm.tsx

## Steps to test
1. checkout branch
2. check homepage for input with an icon and intake page for input without an icon

Screen Shot 2022-08-17 at 9 24 37 PM

## What should reviewers focus on?
* if the component is placed in the right spot
* if the component looks good -> i had to disable eslint to pass in the props using the spread operator (since that's the easiest way to pass props into the Input component without making anything mandatory/missing any props)

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/common/CustomInput.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { InputGroup, Input, InputLeftElement } from "@chakra-ui/react";
+import React from "react";
+
+type CustomInputProps = {
```
Comment (jennchenn) [inline frontend/src/components/common/CustomInput.tsx:4]: typescript allows you to compose a type as as an intersection of types (more info [here](https://www.typescriptlang.org/docs/handbook/2/objects.html#intersection-types))

have you tried importing `ImportProps` here like
`type CustomInputProps = ImportProps & { ....`

i don't think this'll solve the eslint issue, but might help scope down the props we accept?

File: frontend/src/components/common/CustomInput.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  icon?: JSX.Element;
+  [x: string]: unknown;
```
Comment (jennchenn) [inline frontend/src/components/common/CustomInput.tsx:6]: can you explain what this line is for?

File: frontend/src/components/common/CustomInput.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+const CustomInput = React.forwardRef(
+  (
+    { icon, ...props }: CustomInputProps,
+    ref: React.LegacyRef<HTMLInputElement> | undefined,
+  ) => {
+    return (
+      <InputGroup>
+        {icon ? (
```
Comment (jennchenn) [inline frontend/src/components/common/CustomInput.tsx:16]: can this be simplified with [conditional rendering](https://reactjs.org/docs/conditional-rendering.html)?
i.e.
```
{icon && 
     <InputLeftElement pointerEvents="none">{icon}</InputLeftElement>
}

**Review by jennchenn — CHANGES_REQUESTED** [review]
thoughts on removing the `ref` and using the normal `Input` component for court order referral attachment? it doesn't really fall into the regular text input category, and that'll help simplify the `CustomInput` component code

File: frontend/src/components/common/CustomInput.tsx (hunk 2 above)
Comment (juthikahoque) [reply frontend/src/components/common/CustomInput.tsx:6]: this is for the ...props paramater

**Review by jennchenn — APPROVED** [review]
nice!

**Pull request merged**